server/accounts.go

```diff
@@ -4223,20 +4236,38 @@ func (s *Server) fetch(res AccountResolver, name string, timeout time.Duration)
 	respC := make(chan []byte, 1)
 	accountLookupRequest := fmt.Sprintf(accLookupReqSubj, name)
 	s.mu.Lock()
+	// Resolver will wait for available routes + gateways to reply
+	// before serving an error in case there weren't any found.
+	expectedServers := len(s.routes)
+	if s.gateway != nil {
+		expectedServers += len(s.gateway.remotes)
+	}
 	if s.sys == nil || s.sys.replies == nil {
 		s.mu.Unlock()
 		return _EMPTY_, fmt.Errorf("eventing shut down")
 	}
 	replySubj := s.newRespInbox()
 	replies := s.sys.replies
+
 	// Store our handler.
-	replies[replySubj] = func(sub *subscription, _ *client, _ *Account, subject, _ string, msg []byte) {
+	replies[replySubj] = func(sub *subscription, _ *client, _ *Account, subject, reply string, msg []byte) {
 		clone := make([]byte, len(msg))
 		copy(clone, msg)
+
 		s.mu.Lock()
```

Comment:
Could have defer unlock here based on flow, or could grab ok from replies, release lock then do select and placement on the channel.